### PR TITLE
Fix CHANGELOG formatting issues

### PR DESCRIPTION
**Pull request by JulioAPeraza**
Closes None.

Changes proposed in this pull request:

- Fix the admonitions that were reformatted by update-changelog Action. 
- Remove bullet points from multiline elements by having everything on a single line.
- Reorder the release versions by switching the latest version 0.0.12rc6 with 0.0.12rc3. 
- Add links to the full changelog to the headings to match the format of the update-changelog Action.

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -1,9 +1,6 @@
 name: "Run Tests"
 
 on:
-  push:
-    branches:
-      - "main"
```
Comment (tsalo) [inline .github/workflows/testing.yml:6]: Is this a change you want merged?

**Review by tsalo — COMMENTED** [review]
Thanks for fixing the changelog. I think we need to add some info about making releases and/or updating the changelog to the contributing guidelines, but that can come in a later PR.

File: .github/workflows/testing.yml (hunk 1 above)
Comment (JulioAPeraza) [reply .github/workflows/testing.yml:6]: I deleted these lines to skip the test when the CHANGELOG is updated (there is a push to main). Do you think this is the best way to do that? I believe we don't usually have direct commits pushes to main, and the Docs build is not affected by skipping those tests.

Comment (tsalo) [reply .github/workflows/testing.yml:6]: I'm thinking there might be a way to filter based on tags or the commit message/PR title, so we could run it generally, but skip commits from the changelog Action.

**Comment by tsalo** [comment]
Ah, we _do_ have the skip CI action, so if we can include any of the following in the changelog commits, that would work: `[skip ci];[ci skip];[skip github]`.

**Comment by JulioAPeraza** [comment]
That's a very nice solution. Thanks!

**Comment by tsalo** [comment]
Darn, it looks like there's a problem in the YAML file: https://github.com/neurostuff/NiMARE/actions/runs/2490490055/workflow

**Pull request merged**